app/vmauth/main.go

```diff
@@ -200,10 +200,13 @@ func processRequest(w http.ResponseWriter, r *http.Request, ui *UserInfo) {
 		up, hc = ui.DefaultURL, ui.HeadersConf
 		isDefault = true
 	}
-	maxAttempts := up.getBackendsCount()
-	r.Body = &readTrackingBody{
-		r: r.Body,
+	if r.ContentLength <= int64(maxRequestBodySizeToRetry.IntN()) {
```

Comment:
@shichanglin5 have you observed CPU usage reduction after using sync pool?
What's your typical workload for vmauth: read, write or read&write queries?

Reply:
I have separated the read and write workload, where `vmauth-apm` is used for writing, and `vmauth-meta` is used for querying. I have just downgraded `vmauth` to the previous version (without using the reader pool). Below is a comparison of the monitoring data after the downgrade.

**vmauth-apm (for write request)**
<img width="785" alt="image" src="https://github.com/VictoriaMetrics/VictoriaMetrics/assets/48707349/9d2e6bf7-1bbc-46e2-9dee-b43c0a553d92">
<img width="1495" alt="image" src="https://github.com/VictoriaMetrics/VictoriaMetrics/assets/48707349/1744933f-4402-4c1a-9dba-6bdf165f9d1c">


**vmauth-meta (for read request)**
<img width="770" alt="image" src="https://github.com/VictoriaMetrics/VictoriaMetrics/assets/48707349/65f20c41-c81c-4fe4-9182-85ba19c4df45">
<img width="1502" alt="image" src="https://github.com/VictoriaMetrics/VictoriaMetrics/assets/48707349/330480c7-f090-44ec-9487-f9f8daa1acf4">

Reply:
RSS Mem 
<img width="1499" alt="image" src="https://github.com/VictoriaMetrics/VictoriaMetrics/assets/48707349/0c81d9d0-26f7-445f-8db2-40948009d826">